### PR TITLE
feat: Add observer

### LINKED ISSUE #24: Add observer
The Observer watches the public blockchain to identify Sidetree operations, then parses the operations into data structures that can be used for efficient DID resolutions.

### PR DESCRIPTION
**Pull request by fqutishat**
closes #24

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/api/batch/operation.go [hunk 1]
```diff
@@ -32,6 +32,12 @@ type Operation struct {
 	Patch jsonpatch.Patch `json:"patch"`
 	//HashAlgorithmInMultiHashCode
 	HashAlgorithmInMultiHashCode uint `json:"hashAlgorithmInMultiHashCode"`
+	//The logical blockchain time that this operation was anchored on the blockchain
+	TransactionTime uint `json:"transactionTime"`
```
Comment (bstasyszyn) [inline pkg/api/batch/operation.go]: If this is the block number then it should be uint64

File: pkg/observer/observer.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/trustbloc/sidetree-core-go/pkg/docutil"
+)
+
+var logger = logrus.New()
+
+// SidetreeTxn defines info about sidetree transaction
+type SidetreeTxn struct {
+	BlockNumber   uint
```
Comment (bstasyszyn) [inline pkg/observer/observer.go]: uint64

File: pkg/observer/observer.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+					logger.Warnf("Failed to process anchor[%s]: %s", txn.AnchorAddress, err.Error())
+					continue
+				}
+				logger.Debugf("Successfully processed anchor[%s]", txn.AnchorAddress)
+			}
+		}
+	}(sidetreeTxnChannel)
+}
+
+func processSidetreeTxn(sidetreeTxn SidetreeTxn, dacs DCAS, operationStore OperationStore) error {
```
Comment (bstasyszyn) [inline pkg/observer/observer.go]: Typo: should be 'dcas'

File: pkg/observer/observer.go [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+	af, err := getAnchorFile(content)
+	if err != nil {
+		return errors.Wrapf(err, "failed to unmarshal anchor[%s]", sidetreeTxn.AnchorAddress)
+	}
+
+	return processBatchFile(af.BatchFileHash, sidetreeTxn, dacs, operationStore)
+}
+
+func processBatchFile(batchFileAddress string, sidetreeTxn SidetreeTxn, dacs DCAS, operationStore OperationStore) error {
```
Comment (bstasyszyn) [inline pkg/observer/observer.go]: Typo: dcas

Comment (fqutishat) [reply pkg/observer/observer.go]: done

File: pkg/observer/observer.go (hunk 3 above)
Comment (fqutishat) [reply pkg/observer/observer.go]: done

File: pkg/observer/observer.go (hunk 2 above)
Comment (fqutishat) [reply pkg/observer/observer.go]: done

File: pkg/api/batch/operation.go (hunk 1 above)
Comment (fqutishat) [reply pkg/api/batch/operation.go]: done

**Pull request merged**